backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/TimeMachine.java

```diff
@@ -0,0 +1,119 @@
+package de.bund.digitalservice.ris.norms.application.service;
+
+import de.bund.digitalservice.ris.norms.application.service.exceptions.XmlProcessingException;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * Namespace for business Logics related to "time machine" functionality, i.e. to applying LDML.de
+ * "modifications" to LDML.de files. For details on LDML.de modifications, cf. <a
+ * href="https://gitlab.opencode.de/bmi/e-gesetzgebung/ldml_de/-/tree/main/Spezifikation?ref_type=heads">LDML-details</a>
+ */
+@Service
+public class TimeMachine {
+
+  XmlDocumentExtractor xmlDocumentExtractor;
+
+  public TimeMachine(XmlDocumentExtractor xmlDocumentExtractor) {
+    this.xmlDocumentExtractor = xmlDocumentExtractor;
+  }
+
+  /**
+   * Applies the modifications of the amending law onto the target law.
+   *
+   * @param amendingLawString a Document that contains LDML.de modifications to be applied on the
+   *     target law
+   * @param targetLawString The Document that the modifications will be applied to
+   * @return the Document that results in applying the amending law's modifications to the target
+   *     law
+   */
+  public String apply(final String amendingLawString, final String targetLawString) {
+
+    final Document amendingLaw = stringToXmlDocument(amendingLawString);
+    Document targetLaw = stringToXmlDocument(targetLawString);
+
+    final Node firstModificationNodeInAmendingLaw =
+        xmlDocumentExtractor.getFirstModification(amendingLaw);
+
+    targetLaw = applyOneSubstitutionModification(firstModificationNodeInAmendingLaw, targetLaw);
+
+    return convertDocumentToString(targetLaw);
+  }
+
+  /**
+   * Applies one substitution modification
+   *
+   * @param modificationNode the node containing the substitution
+   * @param targetLaw the document version of the target law
+   * @return the <code>targetLaw</code> with the applied modification
+   */
+  private Document applyOneSubstitutionModification(Node modificationNode, Document targetLaw) {
+    final Document targetLawClone = xmlDocumentExtractor.cloneDocument(targetLaw);
+
+    XmlDocumentExtractor.ReplacementPair replacementText =
+        xmlDocumentExtractor.getReplacementText(modificationNode);
```

Comment:
Maybe have the function name similar to the type we're getting.
`getReplacementPair` or `getReplacementTextPair` (if the latter, then maybe also the type `XmlDocumentExtractor.ReplacementTextPair`